modules/table/src/integrationTest/java/org/apache/ignite/internal/disaster/ItDisasterRecoveryReconfigurationTest.java

```diff
@@ -210,6 +213,42 @@ void testManualRebalanceIfMajorityIsLost() throws Exception {
         assertThat(errors, is(empty()));
     }
 
+    /**
+     * Tests that a situation from the test {@link #testInsertFailsIfMajorityIsLost()} it is possible to recover specified partition using
+     * a disaster recovery API.
+     */
+    @Test
+    @ZoneParams(nodes = 5, replicas = 3, partitions = 2)
+    void testManualRebalanceIfMajorityIsLostSpecifyPartitions() throws Exception {
+        int partId = 0;
+        int secondPartId = 1;
+
+        IgniteImpl node0 = cluster.node(0);
+        Table table = node0.tables().table(TABLE_NAME);
+
+        awaitPrimaryReplica(node0, secondPartId);
+
+        stopNodesInParallel(2, 4);
+
+        waitForScale(node0, 3);
+
+        CompletableFuture<?> updateFuture = node0.disasterRecoveryManager().resetPartitions(
+                zoneName,
+                QUALIFIED_TABLE_NAME,
+                Set.of(secondPartId)
+        );
+
+        assertThat(updateFuture, willCompleteSuccessfully());
+
+        awaitPrimaryReplica(node0, secondPartId);
+
+        List<Throwable> fixedPartErrors = insertValues(table, secondPartId, 0);
+        assertThat(fixedPartErrors, is(empty()));
+
+        List<Throwable> secondPartErrors = insertValues(table, partId, 0);
```

Comment:
Please add a comment detailing that this is to make sure that only the partition that was requested is reset